desktop-widgets/command_edit.cpp

```diff
@@ -248,6 +249,29 @@ DiveField EditWaterTemp::fieldId() const
 	return DiveField::WATER_TEMP;
 }
 
+// ***** Atmospheric pressure *****
+void EditAtmPress::set(struct dive *d, int value) const
+{
+	divecomputer *dc = get_dive_dc(d,dc_number); 
+	dc->surface_pressure.mbar = value > 0 ? (uint32_t)value : 0u;
+	fixup_surface_pressure(d);  // create new dive->surface-pressure
```

Comment:
Note that this has strange semantics if you have more than one DC:
You display the average pressure of all DCs, but edit only the current DC.

That means to actually set a given value, the user has to flip through all DCs.

Reply:
Given that dive->surface_pressure is a calculated value and not an original data item, this is the only way of doing it in practice. There are a few others that logically should behave in a similar way, e.g. airtemp which also supplied by the dive computer. This would mean that, when viewing the Information tab, at least airtemp, watertemp and surface_pressure represent the values of the currently-shown dive computer. I need to indicate that in the user manual. They all appear to have a similar logic with equivalent fixup() functions ind dive.c. I hope my interpretation is correct.

Reply:
I think that air-temperature currently behaves fundamentally different: This is saved with the dive. Also note that `fixup_airtemp` only acts if the temperature for the dive is not yet set, so a user-provided value will not be overwritten. Finally, it only seems to use the temperature of the first dive computer, not an average.

I see different ways of handling this:
1) Making pressure behave more like temperature, i.e. save it to the file and don't overwrite it with dive-computer values if the user changed it.
2) Show a non-editable pressure average over all dive computers and an per-divecomputer editable value.

Since I wouldn't dive with more than one DC (Segal's law 😁), I don't have an opinion on the matter.

Reply:
I would definitely prefer the first option.  This would mean that the dc-specific values are not viewable if more than one dc is used. However, dc-specific data are still stored in the XML and are, in principle available to be processed.

Reply:
The code has been changed so that: If the used edits surface pressure in the Information tab, it is saved as dive.surface_pressure. If the dive is saved it is stored in XML as a dive-related attribute. Upon initialising a dive from XML, if dive.surface_pressure has a value, this is used and not calculated from the dive computer. The use case for editing the surface pressure is for dives at altitude when the dive computer does not measure atmospheric pressure. In the corner case when the dive computer does measure pressure and the user modifies the pressure, this is stored and saved as dive.surface_pressure, but the pressure derived from dive computer(s) can be recovered and stored in dive.surface_pressure by typing a zero-value in the text box in the Information tab.